esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -3354,11 +3354,30 @@ bool CWsWorkunitsEx::onWUProcessGraph(IEspContext &context,IEspWUProcessGraphReq
             throw MakeStringException(ECLWATCH_CANNOT_OPEN_WORKUNIT,"Cannot open workunit %s.",wuid.str());
         ensureWsWorkunitAccess(context, *cw, SecAccess_Read);
 
-        Owned <IConstWUGraph> graph = cw->getGraph(req.getName());
-        Owned <IPropertyTree> xgmml = graph->getXGMMLTree(true); // merge in graph progress information
+        unsigned count = cw->getGraphCount();
+        if (count < 1)
+            throw MakeStringException(ECLWATCH_GRAPH_NOT_FOUND, "No graph found in %s.",wuid.str());
+
+        if (isEmpty(req.getName()) && (count > 1))
```

Comment:
So someone making a call without specifying a graph name doesn't know if it will fail or not - it will fail if there are multiple (or zero) graphs, but succeed if there is exactly one? That is not a useful semantic for an API function. It's fine for the user interface to be 'helpful' in this way but the API should be consistent about whether it requires a graph name (i.e. it always should).

Your current code is executing a getGraphCount() function (which is not free - indeed it's actually quite expensive as currently implemented) for no good purpose, other than to allow sloppily written caller code to SOMETIMES get away without passing in a graph name. By the time said caller knows whether it's ok to call without a graph name they know the name of the graph they should supply.

If a graph name is not supplied you should fail.